src/index.js

```diff
@@ -9,40 +10,51 @@ const {
 } = require("./licenseUtils");
 const { getOptions } = require("./optionsUtils");
 
+async function buildAsset(compilation, options) {
+  const { filter, allow, ignore, override, emitError, outputFilename, outputWriter } = options;
+
+  let licenseInformation = getLicenseInformationForCompilation(compilation, filter);
+  licenseInformation = ignoreLicenses(licenseInformation, ignore);
+  licenseInformation = overrideLicenses(licenseInformation, override);
+
+  const licenseViolations = getLicenseViolations(licenseInformation, allow);
+  if (emitError) {
+    compilation.errors.push(...licenseViolations);
+  } else {
+    compilation.warnings.push(...licenseViolations);
+  }
+
+  const sortedLicenseInformation = getSortedLicenseInformation(licenseInformation);
+  return [
+    outputFilename,
+    new RawSource(await writeLicenseInformation(outputWriter, sortedLicenseInformation))
+  ];
+}
+
 class LicenseCheckerWebpackPlugin {
   constructor(options) {
     this.options = getOptions(options);
   }
 
   apply(compiler) {
-    const {
-      filter,
-      allow,
-      ignore,
-      override,
-      emitError,
-      outputFilename,
-      outputWriter
-    } = this.options;
-
-    compiler.hooks.emit.tapPromise("LicenseCheckerWebpackPlugin", async compilation => {
-      let licenseInformation = getLicenseInformationForCompilation(compilation, filter);
-      licenseInformation = ignoreLicenses(licenseInformation, ignore);
-      licenseInformation = overrideLicenses(licenseInformation, override);
-
-      const licenseViolations = getLicenseViolations(licenseInformation, allow);
-      if (emitError) {
-        compilation.errors.push(...licenseViolations);
-      } else {
-        compilation.warnings.push(...licenseViolations);
-      }
-
-      const sortedLicenseInformation = getSortedLicenseInformation(licenseInformation);
-      // eslint-disable-next-line require-atomic-updates
-      compilation.assets[outputFilename] = new RawSource(
-        await writeLicenseInformation(outputWriter, sortedLicenseInformation)
-      );
-    });
+    const name = "LicenseCheckerWebpackPlugin";
+    if (webpack.version.startsWith("4.")) {
```

Comment:
I think this condition is not good, because when `webpack@3` or lower is used, the `else` branch will be used.

Reply:
The peer depend was already `"webpack": "^4.0.0"` and the last `3.x` release was three years ago. That said, this should probably land with a minor version bump to ensure it's a deliberate upgrade for any current users.

Reply:
Could we check for the presence of `thisCompilation` on `compiler.hooks` instead? _(I'm assuming `compiler.hooks.thisCompilation` wasn't a thing before Webpack v5.)_

Reply:
Possibly. I'm going from the chain of discussion at https://github.com/webpack/webpack/issues/11425#issuecomment-690368580 where @jeffposnick said he's prefer to duck type, but ended up landing code that does this fork between the v4 and v5 cases.

Reply:
Ah, aligning with what others are doing sounds sensible. I'm ok with either direction then.